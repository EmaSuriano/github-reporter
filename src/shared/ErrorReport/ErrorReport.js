import React from "react";
import { string } from "prop-types";

import { DEFAULT_ERROR } from "../constants";

const ErrorReport = ({ description }) => (
  <div className="error-box">{description}</div>
);

ErrorReport.defaultProps = {
  description: DEFAULT_ERROR
};

ErrorReport.propTypes = {
  description: string
};

export default ErrorReport;
