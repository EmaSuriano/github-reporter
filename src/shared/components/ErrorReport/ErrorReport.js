import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';
import { DEFAULT_ERROR } from '../../constants';

const ErrorReport = ({ description }) => <div>{description}</div>;

ErrorReport.defaultProps = {
  description: DEFAULT_ERROR,
};

ErrorReport.propTypes = {
  description: string,
};

export default ErrorReport;
