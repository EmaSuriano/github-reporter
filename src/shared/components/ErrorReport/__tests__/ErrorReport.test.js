import React from 'react';
import { shallow } from 'enzyme';

import ErrorReport from '../ErrorReport';

describe('Error Report Component', () => {
  let wrapper;

  describe('when have a default property', () => {
    beforeEach(() => {
      wrapper = shallow(<ErrorReport />);
    });

    it('should render accordingly to the snapshot', () => expect(wrapper).toMatchSnapshot());
  });

  describe('when contains a defined property', () => {
    beforeEach(() => {
      wrapper = shallow(<ErrorReport description="Some Description" />);
    });

    it('should render accordingly to the snapshot', () => expect(wrapper).toMatchSnapshot());
  });
});
