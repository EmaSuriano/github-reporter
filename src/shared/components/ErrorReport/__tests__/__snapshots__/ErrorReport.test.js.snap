// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Report Component when contains a defined property should render accordingly to the snapshot 1`] = `
<div
  className="error-box"
>
  Some Description
</div>
`;

exports[`Error Report Component when have a default property should render accordingly to the snapshot 1`] = `
<div
  className="error-box"
>
  No results found
</div>
`;
